### ISSUE TITLE
Mount root filesystem and USB sticks read-only

### ISSUE THREAD

**Issue by DeviousPenguin** [pideck/pideck#22]
Just a quick question, I was wondering if there is a way of shutting down Pideck cleanly?

Or is the filesystem mounted as readonly so the SD card will be unaffected by powering down?

**Comment by obbardc** [comment]
We are currently not shutting down cleanly. Eventually we hope to mount the SD card read only.

Thanks for reporting this, I probably would have forgotten!

**Comment by hills** [comment]
Yup, better to make it just not necessary to shut down cleanly at all. With everything mounted read-only there's nothing to flush.

**Comment by danielhjames** [comment]
A read-only root filesystem would also help during accidental power losses, as unlike laptop-based systems the PiDeck currently has no battery. It would be good if we could do without a battery; the reboot time is pretty quick at present, and could probably be made shorter.

**Comment by danielhjames** [comment]
These shutdown buttons look good: https://mausberry-circuits.myshopify.com/collections/frontpage/products/illuminated-led-shutdown-switch

**Comment by danielhjames** [comment]
We should also mount the USB sticks read-only, for example in usbmount.conf using the options:
MOUNTOPTIONS="ro,noexec,nodev,noatime,nodiratime"

**Comment by danielhjames** [comment]
I tested these mountoptions for the USB stick and they work just fine. The noatime,nodiratime options are probably redundant in read-only mode, but it might be useful to leave them in case we mount read-write in future, for example if recording.
